dsp-user/interface/src/test/scala/dsp/user/route/CreateUserSpec.scala

```diff
@@ -0,0 +1,153 @@
+                        "password": "test",
+                        "language": "en",
+                        "status": true,
+                        "systemAdmin": false
+                      }""".stripMargin)
+                  )
+        error <- CreateUser.route(request, userHandler).map(_.body).exit
+        _      = println(error)
+      } yield assert(error)(
+        fails(equalTo(ValidationException(s"List(Email is invalid.)")))
+      )
+    },
+    test("return all errors when the payload contains multiple invalid data") {
+      for {
+        userHandler <- ZIO.service[UserHandler]
+        request = Request(
+                    version = Version.`HTTP/1.1`,
+                    method = Method.POST,
+                    url = URL(!! / "admin" / "users"),
+                    headers = Headers.empty,
+                    body = Body
+                      .fromString("""{
+                        "givenName": "Hans",
+                        "familyName": "Muster",
+                        "username": "hansmuster",
+                        "email": "hans.musterexample.org",
+                        "password": "test",
+                        "language": "ch",
+                        "status": true,
+                        "systemAdmin": false
+                      }""".stripMargin)
+                  )
+        error <- CreateUser.route(request, userHandler).map(_.body).exit
+        _      = println(error)
+      } yield assert(error)(
+        fails(equalTo(ValidationException(s"List(Email is invalid., LanguageCode 'ch' is invalid.)")))
+      )
+    },
+    test("successfully create a user") {
+      for {
+        userHandler <- ZIO.service[UserHandler]
+        request = Request(
+                    version = Version.`HTTP/1.1`,
+                    method = Method.POST,
+                    url = URL(!! / "admin" / "users"),
+                    headers = Headers.empty,
+                    body = Body
+                      .fromString("""{
+                        "givenName": "Hans",
+                        "familyName": "Muster",
+                        "username": "hansmuster",
+                        "email": "hans.muster@example.org",
+                        "password": "test",
+                        "language": "en",
+                        "status": true,
+                        "systemAdmin": false
+                      }""".stripMargin)
+                  )
+        response    <- CreateUser.route(request, userHandler).map(_.body)
+        uuid        <- UuidGeneratorTest.getCreatedUuids.head
+        responseStr <- response.asString
+      } yield assertTrue(
+        responseStr.startsWith(
+          s"""{"id":"http://rdfh.ch/users/$uuid","givenName":"Hans","familyName":"Muster","username":"hansmuster","email":"hans.muster@example.org","password":"$$2a"""
+        ) &&
+          responseStr.endsWith(
+            """"language":"en","status":true}"""
```

Comment:
is there a assertion to match a regex? that would be slightly more elegant.  
Otherwise I'd suggest comparing the `.asStringPretty` (or what it's actually called) to a nice-looking json string.

On a different note: how concerned should I be that the response to the "create user" request returns the password hash? (I know it's not productive code, but just to future-proof it)

Reply:
About your concern: you are certainly right (Ivan also pointed this out to me), but that would IMO require a proper API design which requires a definition of our information objects which requires a valid domain model... It's out of scope here.

Reply:
About the ugly way I do the test: I agree that it should be improved. Unfortunately, I need to close this PR as soon as possible because other tasks with high priority are waiting. So, I leave it for now.